pandas/core/indexes/multi.py

```diff
@@ -1189,11 +1189,11 @@ def to_frame(self, index=True, name=None):
         else:
             idx_names = self.names
 
-        result = DataFrame({(name or level):
-                            self._get_level_values(level)
-                            for name, level in
-                            zip(idx_names, range(len(self.levels)))},
-                           copy=False)
+        result = DataFrame(
```

Comment:
Doesn't this lose CategoricalDtype info?

Can you ensure we have a test for

```python
In [12]: idx = pd.MultiIndex.from_product([['a', 'b'], pd.Categorical(['A', 'B'])])

In [13]: idx.to_frame().dtypes
Out[13]:
0      object
1    category
dtype: object

```

and make sure that doesn't regress?

Reply:
Any reason not to just use OrderedDict here? Would satisfy both the guaranteed ordering and dtype fidelity.

Reply:
Seems reasonable to me.